<?xml version="1.0"?>
<launch>
  <arg name="robot_name" default="default_robot"/>
  <arg name="target" default="localhost"/>
  <arg name="init_pose" default=""/>
  <arg name="model" default="$(find s4_description)/xacro/s4_robot.xacro"/>

  <group ns="$(arg robot_name)">
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(arg model) robot_name:=$(arg robot_name)" />
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" >
      <param name="tf_prefix" value="$(arg robot_name)"/>
    </node>
    <group ns="device/comport">
      <arg name="device_name" default="/dev/CANadapter"/>
      <node name="s4_hardware_canlink_to_seriallink" pkg="s4_hardware" type="s4_hardware_canlink_to_seriallink" />
      <node name="s4_hardware_seriallink_decoder"    pkg="s4_hardware" type="s4_hardware_seriallink_decoder" />
      <node machine="$(arg target)" name="s4_hardware_serialport" pkg="s4_hardware" type="s4_hardware_serialport">
        <param name="device_name" value="$(arg device_name)"/>
      </node>
      <node name="s4_hardware_seriallink_encoder"    pkg="s4_hardware" type="s4_hardware_seriallink_encoder" />
      <node name="s4_hardware_seriallink_to_canlink" pkg="s4_hardware" type="s4_hardware_seriallink_to_canlink" />
    </group>
  </group>
</launch>
